features/storage/kvstore/securestore/SecureStore.cpp

```diff
@@ -195,10 +190,8 @@ int SecureStore::set_start(set_handle_t *handle, const char *key, size_t final_d
         return MBED_ERROR_INVALID_ARGUMENT;
     }
 
-    *handle = static_cast<set_handle_t>(_inc_set_handle);
-    ih = reinterpret_cast<inc_set_handle_t *>(*handle);
-
     _mutex.lock();
+    *handle = (_opaque_set_handle*)_ih;
```

Comment:
Removing the local `ih` variable might have a code size impact. I would guess negative, but maybe not.  Worth checking.

Use C++ casts - I think this should be `reinterpret_cast<set_handle_t>(_ih)`. Not quite sure why `static_cast` worked in the original. Because it was opaque?